src/main/scala/cromwell/webservice/PerRequest.scala

```diff
@@ -103,6 +103,14 @@ object PerRequest {
 trait PerRequestCreator {
   implicit def actorRefFactory: ActorRefFactory
 
-  def perRequest(r: RequestContext, props: Props, message: AnyRef, timeout: Duration = 1 minutes) =
-    actorRefFactory.actorOf(Props(new WithProps(r, props, message, timeout)))
+  def perRequest(r: RequestContext,
+                 props: Props, message: AnyRef,
+                 timeout: Duration = 1 minutes,
+                 name: String = PerRequestCreator.endpointActorName) = {
+    actorRefFactory.actorOf(Props(new WithProps(r, props, message, timeout, name)), name)
+  }
+}
+
+object PerRequestCreator {
+  def endpointActorName = "Endpoint-" + java.lang.Thread.currentThread.getStackTrace()(1).getMethodName + System.nanoTime()
```

Comment:
The `getStackTrace()(1).getMethodName` looks a bit hacky but if it's the best way maybe a small comment would be nice.